api/addon/v1alpha1/addon_types.go

```diff
@@ -381,3 +381,30 @@ func (a *Addon) CalculateChecksum() string {
 func (a *Addon) GetInstallStatus() ApplicationAssemblyPhase {
 	return a.Status.Lifecycle.Installed
 }
+
+func (p ApplicationAssemblyPhase) Completed() bool {
```

Comment:
You need unit-tests for every function

Reply:
9f5e34b986ed223f8aa974644b1a6e58707a6e85